### PR TITLE
General CI improvements

### PR DESCRIPTION
**Pull request by chaconpiza**

### CODE REVIEW

File: ci.py [hunk 1]
```diff
@@ -179,7 +182,7 @@ def set_log_dir():
 
 
 def get_changed_files():
-    commit_range = os.environ.get('TRAVIS_COMMIT_RANGE', None)
+    commit_range = travis_commit_range
```
Comment (adriancz) [inline ci.py:185]: I think you no need to create new variable in this case

File: ci.py [hunk 2]
```diff
@@ -225,16 +228,17 @@ def get_message_tags():
...
 
 def get_dirty_modules(dirty_files):
-    LOG.info('dirty_files {0}'.format(dirty_files))
     dirty = set()
     for f in dirty_files:
         if os.path.sep in f:
             mod, _ = f.split(os.path.sep, 1)
 
             if not os.path.exists(os.path.join(mod, 'Dockerfile')):
+                LOG.debug("if1 {0}")
```
Comment (adriancz) [inline ci.py:237]: ?

File: ci.py [hunk 3]
```diff
@@ -225,16 +228,17 @@ def get_message_tags():
...
                 continue
 
             if not os.path.exists(os.path.join(mod, 'build.yml')):
+                LOG.debug("if1 {0}")
```
Comment (adriancz) [inline ci.py:241]: ?

File: ci.py [hunk 4]
```diff
@@ -264,29 +268,52 @@ def get_dirty_for_module(files, module=None):
...
+                LOG.error('BUILD FAILED !!!')
+                raise TempestTestFailedException() #TODO add build exc
+            if return_code == 0:
+                LOG.info('Build succeeded')
+            # Process has finished, read rest of the output 
+            for output in p.stdout.readlines():
+                LOG.debug(output.strip())
+            break
+        end_time = start_time + datetime.timedelta(minutes=20)
+        if datetime.datetime.now() >= end_time:
```
Comment (adriancz) [inline ci.py:299]: You can replace this line wiht 
```suggestion
        if datetime.datetime.now() - start time >= datetime.timdelta(minutes(20):
```
and remove line 298. 
I think is easier to understated this line, but both option are good.

File: ci.py [hunk 5]
```diff
@@ -264,29 +268,52 @@ def get_dirty_for_module(files, module=None):
...
+            if return_code != 0:
+                LOG.error('BUILD FAILED !!!')
+                raise TempestTestFailedException() #TODO add build exc
+            if return_code == 0:
+                LOG.info('Build succeeded')
+            # Process has finished, read rest of the output 
+            for output in p.stdout.readlines():
+                LOG.debug(output.strip())
+            break
+        end_time = start_time + datetime.timedelta(minutes=20)
```
Comment (adriancz) [inline ci.py:298]: I recommend to create 'TIMEOUT' variable, we should avoid magic numbers.

File: ci.py [hunk 6]
```diff
@@ -602,7 +631,7 @@ def wait_for_init_jobs(pipeline):
     docker_id_dict = {job: "" for job in INIT_JOBS[pipeline]}
 
     amount_succeeded = 0
-    for attempt in range(40):
+    for attempt in range(5):
```
Comment (adriancz) [inline ci.py:634]: Maybe is good idea to add variable like attempt_limit for this number.

File: ci.py [hunk 7]
```diff
@@ -199,7 +202,7 @@ def get_changed_files():
 
 
 def get_message_tags():
-    commit = os.environ.get('TRAVIS_COMMIT_RANGE', None)
+    commit = travis_commit_range
```
Comment (adriancz) [inline ci.py:205]: Pleas check the comment above.

File: ci.py [hunk 8]
```diff
@@ -808,100 +837,99 @@ def kill(signal, frame):
...
             # Process has finished, read rest of the output 
-            #for output in p.stdout.readlines():
-            #    LOG.error(output.strip())
+            for output in p.stdout.readlines():
+                LOG.debug(output.strip())
             break
         end_time = start_time + datetime.timedelta(minutes=20)
         if datetime.datetime.now() >= end_time:
-            LOG.error('Tempest-tests timed out at 20 min !!!')
+            LOG.error('TEMPEST-TEST TIMEOUT AFTER 20 MIN !!!')
```
Comment (adriancz) [inline ci.py:850]: Pleas check previous comment

**Review by adriancz — CHANGES_REQUESTED** [review]

File: ci.py (hunk 2 above)
Comment (chaconpiza) [reply ci.py]: Done

File: ci.py (hunk 3 above)
Comment (chaconpiza) [reply ci.py]: Done

File: ci.py (hunk 5 above)
Comment (chaconpiza) [reply ci.py]: Done

File: ci.py (hunk 6 above)
Comment (chaconpiza) [reply ci.py]: Done

File: ci.py (hunk 7 above)
Comment (chaconpiza) [reply ci.py]: Done

File: ci.py (hunk 8 above)
Comment (chaconpiza) [reply ci.py]: Done

File: ci.py (hunk 4 above)
Comment (chaconpiza) [reply ci.py:303]: Then let's leave with the `end_time` var

File: ci.py (hunk 1 above)
Comment (chaconpiza) [reply ci.py]: done

**Review by adriancz — APPROVED** [review]

**Pull request merged**